src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,135 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
+@SuppressWarnings("unchecked")
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final Object[] EMPTY_ELEMENT_DATA = {};
+    private Object[] elementList;
```

Comment:
You have class type parameter, why use object

Reply:
As i've read through the internet, JAVA doesn't support generic arrays.Anyway i've some examples like https://www.baeldung.com/java-generic-array, but for me using Object is better for code readability, also i noticed that ArrayList "default" method also uses Object.

Reply:
we can't create generic array using new, but we can declare it, assign array of objects to it and cast